### ISSUE TITLE
Enhancement: Varargs in assertThat and method isOnlyValueIn and multiGet of List

### ISSUE THREAD

**Issue by mtravnicek** [joel-costigliola/assertj-core#1466]
I have test which checks selected rows for "status green". After some operation say 3 distinct rows are green.
I have `List<String> allRows ` with row values
test so far is checking:
```
allRows.get(1).contains(status_green) &&
allRows.get(3).contains(status_green) &&
allRows.get(4).contains(status_green) == true
```
So to remove redundancy, i can do something like:
`asssertThat(status_green).isIn(allRows.get(1),allRows.get(3),allRows.get(4))`
but this is **wrong** because this would have to be **isOnlyValueIn** (can this be added?)
i can reverse it into ugly but working:
`assertThat(Arrays.asList({allRows.get(1)..get(3)..get(4)}).containsOnly(status_green)`
So is there any chance to have **varargs assertThat** like (example with plain integers) 
`assertThat(1,2,3).containsOnly(1);`
But my problem also lies in those repeated gets. Some **multiGet** would be great. Like 
`assertThat(allRows).get/filter(1,3,4).containsOnly...`
_(This stays in AssertJ domain and I could not find library with List allowing multiGet)_

**Comment by joel-costigliola** [comment]
If I understand correctly, you could use filtering on `allRows` and check the expected content of the filtered list:
```
assertThat(allRows).filteredOn(row -> hasGreenStatus(row))
                   .containsOnly(allRows.get(1), allRows.get(3), allRows.get(4));
```

**Comment by mtravnicek** [comment]
Well that somehow brokes logic of the test.
I have made some action on web, which affects 3 selected rows and i want to check,
that status of these rows changed accordingly...  (status green for example)
I have function `allRows` returning List<String> of values of status of **every** row
and work only with strings. In your example I would have to introduce new metod `hasGreenStatus`.
Above all in your code example no other row is permitted to have green status, which is wrong.
test should not care about other rows...
Of course I can create method which will fetch me only selected row values, and apply assert on those.
And as i mentioned there is also mess with repeated gets.
I would like to add something like this
`assertThat("green").isOnlyValueIn("green","green","red") => fail`
or even 
`assertThat(allRows).get(1,3,4).containsOnly("green") => fail`
I originally thought this is how the **isIn** works!
I can do ofcourse something ugly 3x times longer like:
```
List<Integer> tested_rows = Lists.newArrayList(1,3,4);
List<String >selected_rows =  tested_rows.stream().map(column::get)
            			.collect(Collectors.toList());
assertThat(selectedRows).containsOnly("green");
```
So i would like to shift burden to AssertJ for segmenting and selecting whats tested
at which it is good so far 👍 and not create any helper methods, code myself.
I can even add this method to code myself... (must this be created and customized for all tested containers? or maybe only List/Iterable String suffice?)
You didnt reflect on any of my hints towards this...
Maybe you dont want to have code polluted, but assertThat singlevalue methods are very few.
Sorry for long post, but is so many variations of `assertThat` method necessary?
Cant be object type inferred from Object by getClass() etc.? I think that hamcrest and junit dont have that many constructors... just guessing

**Comment by joel-costigliola** [comment]
I did not understand your test then, it would help me help you if you can provide a complete test case which I could run in my IDE.

`hasGreenStatus` can easily be implemented with a lambda if you don't want an additional method.

> Above all in your code example no other row is permitted to have green status, which is wrong.
test should not care about other rows...

You can use `contains` instead of `containsOnly` in that case but again what you want is not crystal clear to me so I'm might still be wrong.

>  You didnt reflect on any of my hints towards this...

I did not because I don't understand clearly what you are trying to achieve so maybe there is already an assertion that fits your need but I can't tell until you clarify things.

>  Maybe you don't want to have code polluted, but assertThat singlevalue methods are very few.

I'm not sure to understand, can you clarify this comment ?

> Sorry for long post, but is so many variations of assertThat method necessary?

Yes

> Cant be object type inferred from Object by getClass() etc.? 

No, `assertThat(x)` is selected according to the declared type of `x` not its runtime type, this how java chooses which overloaded method is going to be used.

**Comment by mtravnicek** [comment]
> Maybe you don't want to have code polluted, but assertThat singlevalue methods are very few.

With this i was hinting that "reverse" method aka **assertThat(singleValue)** as opposed to **assertThat(list/iterable)** while already having method **isIn** could be extened with method **isOnlyValueIn** and there is few methods of this kind, so it wont be bloated.

My testcase again: i have selenium web test in which:
i have web form on which you select records/rows to perform some operation say **publish**.
**1. select some rows - say (1, 2, 4)**
**2. perform publish**
**3. use function which returns List of statuses of ALL rows.**
**4. check that rows (1,2,4) has status (publish/green) etc.** (dont care about other rows)

So i like to make assert for that as shortest possible and oneliner.
Of course i can somehow get data of rows and afterwards supply this to assert
I just want to shift that work (segmenting, getting value) to AssertJ.
As this is what i see its strenght working directly with lists (iterables).

**Comment by joel-costigliola** [comment]
so basically you want AssertJ to provide `assertThat(Object... actuals)` that behaves like `assertThat(list(actuals))`, right ? (where `list(Object... values)` just returns a list build from the vararg values).

If that is correct, in the meantime you can use the `List<T>  org.assertj.core.util.Lists.list(T... elements)` and write:

```
assertThat(list(allRows.get(1), allRows.get(3), allRows.get(4)).containsOnly(status_green);
```

**Comment by mtravnicek** [comment]
yes, that is correct, varargs List would be great... now what about some
`assertThat(allRows).get(1,3,4).containsOnly(status_green);`
shorter and only needs some multiget selector added, hmm?
or maybe name it "filter"? i know filteredOn is already there and cannot be used in this context,
(too bad index is not property of list record)
so can we create something like this only for Lists?

**Comment by joel-costigliola** [comment]
Just to clarify, you would like both `<T> assertThat(T... actuals)` and `assertThat(allRows).get(Integer... indexes)` or only the the latter one ?

**Comment by mtravnicek** [comment]
Both would be nice... or atleast one of these. I dont know how the process of adding/voting for new features works. Is it ok with you and your opinion?
